### PR TITLE
ENH: Add a ``Bandpass`` filter interface under ``algorithms.filters``

### PR DESCRIPTION
**Pull request by oesteban**
Upranks the `bandpass_filter` implementation that we had within the example workflows to a fully fledged Nipype Interface.

### CODE REVIEW

File: nipype/algorithms/filters.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                        absolute_import)
+
+from ..utils.filemanip import fname_presuffix
+from ..interfaces.base import (traits, TraitedSpec, SimpleInterface,
+                               BaseInterfaceInputSpec, File)
+
+
+class BandpassInputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, mandatory=True, desc='functional data')
+    freq_low = traits.Range(0.0, min=0.0, usedefault=True, desc='low frequency cutoff')
```
Comment (satra) [inline nipype/algorithms/filters.py]: perhaps specify that these are in Hz. (similar to the original desc). and may be we can add the semantics of 0.

freq_hi: the default of 0 sets high pass cutoff to 0
freq_low: the default of 0 sets low pass cutoff to Nyquist

File: nipype/algorithms/filters.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        # "np.round(..." replaced by "int(np.round(..."
+        lowidx = int(np.round(freq_low / sampling_rate * timepoints))
+
+    highidx = 0
+    if freq_hi > 0:
+        highidx = int(np.round(freq_hi / sampling_rate * timepoints))  # same
+
+    F[highidx:lowidx] = 1
+    F = ((F + F[::-1]) > 0).astype(int)
+    data = img.get_data()
```
Comment (satra) [inline nipype/algorithms/filters.py]: this is now `get_fdata()`

**Comment by oesteban** [comment]
Thumbs up?

**Comment by satra** [comment]
@oesteban - this looks good. 

could we add a real test to this perhaps generating a random (white noise) nifti timeseries file and filtering it with this code in _bandpass_filter? 

also could we add a pointer to the discussion on neurostars (or at least the wording about ringing artifacts with this method)?

**Comment by oesteban** [comment]
will do

File: nipype/algorithms/filters.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            output file name
+
+    """
+    import numpy as np
+    import nibabel as nb
+
+    if freq_hi > 0 and freq_low >= freq_hi:
+        raise ValueError("Low-cutoff frequency can't be greater than the high-cutoff")
+
+    img = nb.load(in_file)
```
Comment (effigies) [inline nipype/algorithms/filters.py]: Do we no longer need this?

```suggestion
    img = nb.load(in_file, mmap=NUMPY_MMAP)
```

File: nipype/algorithms/filters.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+
+def _bandpass_filter(in_file, tr=None, freq_low=0, freq_hi=0, out_file=None):
+    """
+    Bandpass filter the input files
+
+    Parameters
+    ----------
+        files : str
+            4D NIfTI file
```
Comment (effigies) [inline nipype/algorithms/filters.py]: Necessarily NIfTI?

File: nipype/algorithms/filters.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        freq_low : float
+            cutoff frequency for the low pass filter (in Hz)
+            the default of 0 sets low pass cutoff to Nyquist
+        freq_hi : float
+            cutoff frequency for the high pass filter (in Hz)
+            the default of 0 sets high pass cutoff to 0
```
Comment (effigies) [inline nipype/algorithms/filters.py]: This seems wrong...

**Comment by effigies** [comment]
@oesteban You're going to need to merge/rebase master to fix the Docker builds.

**Comment by effigies** [comment]
@oesteban Do you want to try to finish this up with Satra's suggestions this weekend, or just push off to another release?

**Comment by effigies** [comment]
This pull request is "orphaned," which means it has been deemed to be abandoned by its original author. Orphaned pull requests have not been rejected, and we hope that if a user sees one that will meet their needs with a little work, that they will fork it and open a new pull request (or, in the case of the original author, reopen the original PR).

We ask that all adopted PRs be updated to merge or rebase the current master. If you would like to adopt a PR and need help getting started, any of a number of contributors will be happy to help.

**Pull request closed without merging**